### ISSUE TITLE
Pulsar Broker Sometimes does not return CommandSendReceipt/CommandSendError to prducer send CommandSend

### ISSUE THREAD

**Issue by alphashaw** [apache/pulsar#5770]
I'm writing a Pulsar client library for Erlang. I first started with the producer implementation.
I'm doing non/partitioned topics publishing  and everything at protocol level works fine. 
However, when I'm doing a batch partitioned topic produce( max. 1000 msgs) with a max. **batch delay** of 1ms, I noticed that some **CommandSend** requests doesn't get an equivalent **CommandSendReceipt** response. The broker never returns that. This is causing one of the producer of the partition topic to stuck. All the producers share the same connection, and I'm using the the prodcuer_id and the sequence_id of each **CommandSend** to assocate with the corresponding **CommandSendReceipt's**. Is the broker shedding off my loads to avoid overloading? I suspect this because when I increase the batch delay everything works fine. However, I'm not sure. Why is this behaviour? If this is expected, is the sequence_id of the responses that make it be in the same order as the sequence_id of the **CommandSend** requests? In other words, does the broker serially handles requests from the all producers (partitioned/not) on the same connection?

**Expected behavior**
If load shedding is not happening, I expects to see **CommandSendReceipt's** or a **CommandSendError** for  any **CommandSend** request.

**Screenshots**
Below is a photo of the command send and the associated receipt command. It shows where the break happens.

![kaja](https://user-images.githubusercontent.com/14175735/69847443-0f903400-126f-11ea-8fe3-7519484a7450.png)

**Desktop.**
 - OS: [e.g. Ubuntu]

**Comment by jiazhai** [comment]
@alphashaw Great to know that you are working on an Erlang client.
In current client, each partition is handled in an individual connection. so for several partitions in broker side the order between partitions is not guaranteed.

**Comment by alphashaw** [comment]
@jiazhai  Thanks for the swift response. It seems that my last questioned wasn't clear semantically. Let me rephrase it: 

> From the client end, I understand multiple producers (partitioned/not) might be using the same underlying TCP connection. Also I understand the **sequence_id** (for necessary commands)  for each producer using the same connection must be unique. My question is, for these client connection-shared producers, does the broker serially handles their requests? In order words, if **p1** and **p2** represents two arbitrary producers sharing the same client connection and **(req1, res1),(req2, res2)** are at some point their request and response pairs respectively, if req1 < req2 in time order, is it guaranteed that res1 < res2 ??

**Comment by jiazhai** [comment]
@alphashaw I think in your example,  **res1 < res2**  is not guaranteed.

**Comment by alphashaw** [comment]
Yeah. I realized that after some few tests. I was using a queue to hold the request metadata to be poll when a responsive arrived. I realized, in some times, the head of the queue does not match the response. Thanks @jiazhai . 
I have a working producer implementation here [Pulserl](https://github.com/alphashaw/pulserl). The README doc is very minimal right now. I'll give it more look over the weekend.

**Comment by tisonkun** [comment]
Closed as stale. Please create a new issue if it's still relevant to the maintained versions.
